src/Common/src/System/Security/Cryptography/DerSequenceReader.cs

```diff
@@ -110,6 +110,67 @@ internal byte[] ReadNextEncodedValue()
             return encodedValue;
         }
 
+        internal byte[] ReadRemainingData()
+        {
+            int remainingBytes = _end - _position;
+            byte[] remainingData = new byte[remainingBytes];
+            Buffer.BlockCopy(_data, _position, remainingData, 0, remainingBytes);
+            return remainingData;
+        }
+
+        private static byte[][] SplitValue(byte[] payload, ref int offset)
+        {
+            int lengthLength;
+            int contentLength = ScanContentLength(payload, offset + 1, out lengthLength);
+
+            byte[] encodedLength = new byte[lengthLength];
```

Comment:
Yes, while talking to Jeremy we agreed that the structure of the DerSequenceReader and DerEncoder need revision. I added some methods to reduce from byte[][][] to byte[][] but the amount of allocations needed are still high.